From c7502ea0edd236e2fed574147ac8fbb9a32140a2 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Sat, 22 Jun 2013 00:39:00 +0530
Subject: [PORT FROM MAIN]Audio-Offload:Fix for seeking multilpe times at EOS

BZ: 79143

This patch will fix BUFFER TIMEOUT issue when sliding progress bar
fast forward/backward at EOS during audio playback.

In File Manager or preview player, flush will keep on getting called
until you take out your finger from progress bar and this causes
mRetryCount to become zero in prepareTracks() and causes BUFFER TIMEOUT.

Change-Id: If7bbf523b2cef80b04f5a365f0b7c2fc614f8fac
Orig-Change-Id: Ia921ad293cab236d55e11be09afbdd5b59416733
Signed-off-by: samaga Prasanna Krishna <samaga.krishna@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/Tracks.cpp |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/services/audioflinger/Tracks.cpp b/services/audioflinger/Tracks.cpp
index fd27b3d..c4ab7cd 100644
--- a/services/audioflinger/Tracks.cpp
+++ b/services/audioflinger/Tracks.cpp
@@ -847,7 +847,10 @@ void AudioFlinger::PlaybackThread::Track::flush()
                 ALOGV("flushed in STOPPING_1 or 2 state, change state to ACTIVE");
                 mState = ACTIVE;
             }
-
+            if (mState == ACTIVE) {
+                ALOGV("flush called in active state, resetting buffer time out retry count");
+                mRetryCount = kMaxTrackRetriesOffloaded;
+            }
             playbackThread->getOutput_l()->stream->flush(
                                       playbackThread->getOutput_l()->stream);
         }
-- 
1.7.4.1

